src/test/java/com/google/sps/servlets/MatchesListServletTest.java

```diff
@@ -0,0 +1,174 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.json.JSONObject;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.List;
+import java.util.Arrays;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Test;
+import org.junit.After;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.Before;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.common.collect.ImmutableList;
+import com.google.gson.Gson;
+
+@RunWith(JUnit4.class)
+public class MatchesListServletTest {
+  private static final String MATCH_INFO_ENTITY = "match-info";
+  private static final String USER_ID_PROPERTY = "id";
+  private static final String POTENTIAL_MATCHES_PROPERTY = "potential-matches";
+  private static final String FRIENDED_IDS_PROPERTY = "friended-ids";
+  private static final String PASSED_IDS_PROPERTY = "passed-ids";
+  private static final String MATCHES_LIST_PROPERTY = "matches-list";
+  private static final String USER_ID_REQUEST_URL_PARAM = "userid";
+
+  //Test user ids
+  private static final String TEST_USER_ID = "5555";
+  private static final String TEST_CONNECTION_1_ID = "1776";
+  private static final String TEST_CONNECTION_2_ID = "1234";
+  private static final String TEST_CONNECTION_3_ID = "9876";
+
+  private static final List<String> EMPTY_LIST = Arrays.asList();
+
+  private final LocalServiceTestHelper helper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Mock
+  private HttpServletRequest mockRequest;
+ 
+  @Mock
+  private HttpServletResponse mockResponse;
+
+  private MatchesListServlet servletUnderTest;
+  private DatastoreService datastore;
+  private StringWriter responseWriter;
+  private PrintWriter writer;
+
+  @Before
+  public void setUp() throws IOException {
+    MockitoAnnotations.initMocks(this);
+
+    helper.setUp();
+
+    servletUnderTest = new MatchesListServlet();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /**
+   * Tests the case in which a user's match list is empty.
+   *
+   * <p>Should return an empty list.
+   */
+  @Test
+  public void matchInfoWithEmptyList() throws IOException{
+    addMatchInfoEntityToDatastore(TEST_USER_ID, ImmutableList.of(),
+      ImmutableList.of(), ImmutableList.of(), ImmutableList.of());
+    
+    String actualOutput = execute(TEST_USER_ID);
```

Comment:
Let's keep the naming/layout simple here:

```
String jsonOutput = execute(TEST_USER_ID);
List<String> matches = Arrays.asList(new Gson().fromJson(jsonOutput, String[].class));
```

Generally (though it's not a hard and fast rule), I like to group my lines of code for a test in three "blocks" where each "block" is separated:

```
@Test
public void niksAwesomeTest() throws Exception {
  testSetUpLineNo1;
  testSetUpLineNo2;
  testSetUpLineNo3;

  runTestLineNo1;
  runTestLineNo2;

  assertResultLineNo1;
  assertResultLineNo2;
  assertResultLineNo3;
}
```

Reply:
Got it!